test/test_spectral_ops.py

```diff
@@ -1167,18 +1164,58 @@ def _test_istft_is_inverse_of_stft(stft_kwargs):
                 'normalized': True,
                 'onesided': False,
             },
-            # hamming_window, not centered, not normalized, onesided
+            # hamming_window, centered, not normalized, onesided
             # window same size as n_fft
             {
                 'n_fft': 5,
                 'hop_length': 2,
                 'win_length': 5,
                 'window': torch.hamming_window(5, dtype=dtype, device=device),
-                'center': False,
+                'center': True,
                 'pad_mode': 'constant',
                 'normalized': False,
                 'onesided': True,
             },
+        ]
+        for i, pattern in enumerate(patterns):
+            _test_istft_is_inverse_of_stft(pattern)
+
+    @onlyOnCPUAndCUDA
+    @skipCPUIfNoFFT
+    @dtypes(torch.double)
+    def test_istft_round_trip_with_padding(self, device, dtype):
+        """long hop_length or not centered may cause length mismatch in the inversed signal"""
+        def _test_istft_is_inverse_of_stft_with_padding(stft_kwargs):
+            # generates a random sound signal for each tril and then does the stft/istft
+            # operation to check whether we can reconstruct signal
+            data_sizes = [(2, 20), (3, 15), (4, 10)]
+            num_trials = 100
+            istft_kwargs = stft_kwargs.copy()
+            del istft_kwargs['pad_mode']
+            for sizes in data_sizes:
+                for i in range(num_trials):
+                    original = torch.randn(*sizes, dtype=dtype, device=device)
+                    stft = torch.stft(original, return_complex=True, **stft_kwargs)
+                    inversed = torch.istft(stft, length=original.size(-1), **istft_kwargs)
```

Comment:
```suggestion
                    with self.assertWarnsOnceRegex(UserWarning, "signal is shorter than the length parameter"):
                        inversed = torch.istft(stft, length=original.size(-1), **istft_kwargs)
```

Reply:
Thanks. It's resolved in https://github.com/pytorch/pytorch/pull/63469/commits/a776165d4fd73beb21b5271ed8bb653e85710e28